tico/experimental/quantization/ptq/observers/base.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) 2025 Samsung Electronics Co., Ltd. All Rights Reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from typing import Optional, Tuple
+
+import torch
+
+from tico.experimental.quantization.ptq.dtypes import DType, UINT8
+from tico.experimental.quantization.ptq.qscheme import QScheme
+
+
+class ObserverBase(ABC):
+    """
+    Minimal abstract base for all observers/quantizers.
+
+    Subclasses must implement:
+      - reset()
+      - collect(x)
+      - fake_quant(x)
+      - compute_qparams(): optional in practice for some observers (e.g., MX),
+        but still part of the interface; those can return None.
+    """
+
+    def __init__(
+        self,
+        *,
+        name: str,
+        dtype: DType = UINT8,
+        qscheme: QScheme = QScheme.PER_TENSOR_ASYMM,
+        channel_axis: Optional[int] = None,  # None → per-tensor
+    ):
+        self.name = name
+        self.dtype = dtype
+        self.qscheme = qscheme
+        self.channel_axis = channel_axis if qscheme.is_per_channel() else None
+        self.enabled = True
+        self.reset()
+
+    @abstractmethod
+    def reset(self) -> None:
+        """Clear any running statistics or cached params."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def collect(self, x: torch.Tensor) -> None:
+        """
+        Update running statistics with a new batch of data.
+
+        Must be implemented by subclasses (e.g., MinMax, EMA, Histogram, MSE).
+        """
+        raise NotImplementedError
```

Comment:
When enabled == False, `collect` supposed to be void function in this design, right? How about putting that implementation here in base?

Reply:
It's optional. AFAIK, Abstract class can still have some implemented functions. 
```py
def collect(...
  if not enabled:
    return
  update_statistics()

@abstractmethod
def update_statistics(...


```

Like above.

This is optional, though